File: 1.3/URLify.java
/*
URLify.java
*/

class URLify {
	public static String URLify(String input, int len) {
		char[] inputArray = input.toCharArray();

		for (int i = 0; i < len; i++) {
			if (inputArray[i] == ' ') {
				for (int j = i + 1; j < input.length() - 2; j++) {
					curr = 
					inputArray[j + 2] = inputArray[j];
				}
				inputArray[i] = '%';
				inputArray[i+1] = '2';
				inputArray[i+2] = '0';
			}
		}
		String result = new String(inputArray);
		return result;
	}

	public static void main(String[] args) {
		String in = "Hello World  ";
		int trueLen = 11;
		System.out.println(URLify(in, trueLen));
	}
}

File: ctci/1.3/URLify.py
def URLify(input, length):
  input = list(input)
  spaceCount = 0
  for i in range(length):
    if input[i] == ' ':
      spaceCount += 1
  newLen = length + (2 * spaceCount)
  result = newLen * [' ']
  for j in range(length - 1, -1, -1):
    if input[j] == ' ':
      result[newLen - 1] = '0'
      result[newLen - 2] = '2'
      result[newLen - 3] = '%'
      newLen -= 3
    else:
      result[newLen - 1] = input[j]
      newLen -= 1
  return ''.join(result)

def main():
  s = "Hello World  "
  trueLen = 11
  print(URLify(s, trueLen))
main()